packages/core/ui/ErrorMessageStackTraceDialog.tsx

```diff
@@ -7,32 +7,62 @@ import {
   Typography,
 } from '@mui/material'
 
-import { RawSourceMap, SourceMapConsumer } from 'source-map-js'
+import { SourceMapConsumer } from 'source-map-js'
 import copy from 'copy-to-clipboard'
 
 // locals
 import Dialog from './Dialog'
 import LoadingEllipses from './LoadingEllipses'
 
+function Link2({
+  href,
+  children,
+}: {
+  href: string
+  children: React.ReactNode
+}) {
+  return (
+    <Link target="_blank" href={href}>
+      {children}
+    </Link>
+  )
+}
+
+async function myfetch(uri: string) {
+  const res = await fetch(uri)
+  if (!res.ok) {
+    throw new Error(`HTTP ${res.status} fetching ${uri}: ${await res.text()}`)
+  }
+  return res
+}
+
+async function myfetchjson(uri: string) {
+  const res = await myfetch(uri)
+  return res.json()
+}
+
+async function myfetchtext(uri: string) {
+  const res = await myfetch(uri)
+  return res.text()
+}
+
 // produce a source-map resolved stack trace
 // reference code https://stackoverflow.com/a/77158517/2129219
-const sourceMaps: Record<string, RawSourceMap> = {}
+const sourceMaps: Record<string, SourceMapConsumer> = {}
 async function getSourceMapFromUri(uri: string) {
   if (sourceMaps[uri] != undefined) {
     return sourceMaps[uri]
   }
   const uriQuery = new URL(uri).search
-  const currentScriptContent = await (await fetch(uri)).text()
+  const currentScriptContent = await myfetchtext(uri)
 
   let mapUri =
     new RegExp(/\/\/# sourceMappingURL=(.*)/).exec(currentScriptContent)?.[1] ||
     ''
   mapUri = new URL(mapUri, uri).href + uriQuery
 
-  const map = await (await fetch(mapUri)).json()
-
+  const map = new SourceMapConsumer(await myfetchjson(mapUri))
```

Comment:
this is a bonus optimizatiom. if the stack trace goes through 20 stack frames of mobx-state-tree function calls, then it was reparsing the stack trace 20 times which added up to 5 seconds of waiting for the stack trace. this parses it just once